hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -369,6 +379,23 @@ protected void completeCompaction(HoodieCommitMetadata metadata, HoodieTable tab
     LOG.info("Compacted successfully on commit {}", compactionCommitTime);
   }
 
+  public void commitLogCompaction(String compactionInstantTime, HoodieWriteMetadata<O> writeMetadata, Option<HoodieTable> tableOpt) {
+    // dereferencing the write dag for log compaction for the first time.
+    List<HoodieWriteStat> writeStats = triggerWritesAndFetchWriteStats(writeMetadata);
```

Comment:
`triggerWritesAndFetchWriteStats` should not be engine-specific.  Should it be implemented based on `HoodieData`?

Reply:
https://github.com/apache/hudi/pull/13236 -> we are going to need this abstraction in this follow up patch.